### ISSUE TITLE
Generic index signature not inferred after using rest operator

### ISSUE THREAD

**Issue by OliverJAsh** [microsoft/TypeScript#50186]
# Bug Report

### 🔎 Search Terms

- index signature
- generic function
- rest operator

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about index signatures

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ssl=14&ssc=1&pln=14&pc=17#code/CYUwxgNghgTiAEYD2A7AzgF3kgXPA3gFDwnwBmSu8KArgLYBGIMA3MaQ7HrY82wL5tCoSLATJ0WMijwAeAIIA+ABQAPPPngBtANYgAnnkwwAligDmAXTzz4-AJTwAvIvgA3JCeBCA9D-iAoOTwAHJI8MwwSDCE0spI9mwSmATklAA08AB02XDJ-M7YbH7wgDLk8ACqKCCqAA7gGCDA4TCR0bG5GAmExUEA6lE6sEg0KMAZKGERUTEoyprZmR12CUA)

### 💻 Code

```ts
declare const o: {
    foo: number;
    bar: number;
};

declare const fn: <A>(x: { [key: string]: A }) => void;

// ✅ No error
fn(o);
const { foo, ...rest } = o;
// ❌ Unexpected error
fn(rest);
// ✅ Workaround, no error
fn({ ...rest });
```

### 🙁 Actual behavior

See code comments above.

### 🙂 Expected behavior

See code comments above.

**Comment by MartinJohns** [comment]
Duplicate of #48014.

**Comment by OliverJAsh** [comment]
Thanks
